src/Control/Monad/Free/Trans.purs

```diff
@@ -76,6 +79,18 @@ instance monadRecFreeT :: (Functor f, Monad m) => MonadRec (FreeT f m) where
         Loop s1 -> go s1
         Done a -> pure a
 
+instance semigroupFreeT :: (Functor f, Monad m, Monoid w) => Semigroup (FreeT f m w) where
```

Comment:
You can relax the `Monad` constraint to `Applicative` and use `append = lift2 append`.